src/main/java/mate/academy/store/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,45 @@
 package mate.academy.store.service.impl;
 
+import jakarta.persistence.EntityNotFoundException;
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.store.dto.BookDto;
+import mate.academy.store.dto.CreateBookRequestDto;
+import mate.academy.store.mapper.BookMapper;
 import mate.academy.store.model.Book;
 import mate.academy.store.repository.BookRepository;
 import mate.academy.store.service.BookService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setAuthor(requestDto.getAuthor());
```

Comment:
You already map Dto to Book here `Book book = bookMapper.toModel(requestDto);`
remove redundant mapping